### ISSUE TITLE
Build fails on i686 when using system ck

### ISSUE THREAD

**Issue by xavierba** [akopytov/sysbench#118]
Hi,

sysbench 1.0.x doesn't build on i686 when using the system ck (tried with both 0.5.2 and 0.6.0).
It builds fine when using the bundled ck (but using bundled libraries is forbidden in Fedora, which is the distro I'm maintaining sysbench for).
It also builds fine with system ck for other arches (x86_64 aarch64 armv7hl)

Here's an extract of the log :
sysbench.o: In function `sb_more_events':
/builddir/build/BUILD/sysbench-1.0.3/src/sysbench.c:696: undefined reference to `ck_pr_faa_64'
sysbench.o: In function `sb_counter_val':
/builddir/build/BUILD/sysbench-1.0.3/src/sb_counter.h:62: undefined reference to `ck_pr_load_64'
sysbench.o: In function `sb_counter_inc':
/builddir/build/BUILD/sysbench-1.0.3/src/sb_counter.h:68: undefined reference to `ck_pr_store_64'
db_driver.o: In function `sb_counter_val':
/builddir/build/BUILD/sysbench-1.0.3/src/sb_counter.h:62: undefined reference to `ck_pr_load_64'
db_driver.o: In function `sb_counter_inc':
/builddir/build/BUILD/sysbench-1.0.3/src/sb_counter.h:68: undefined reference to `ck_pr_store_64'
db_driver.o: In function `sb_counter_val':
/builddir/build/BUILD/sysbench-1.0.3/src/sb_counter.h:62: undefined reference to `ck_pr_load_64'
db_driver.o: In function `sb_counter_inc':
/builddir/build/BUILD/sysbench-1.0.3/src/sb_counter.h:68: undefined reference to `ck_pr_store_64'
db_driver.o: In function `sb_counter_val':
/builddir/build/BUILD/sysbench-1.0.3/src/sb_counter.h:62: undefined reference to `ck_pr_load_64'
db_driver.o: In function `sb_counter_inc':
/builddir/build/BUILD/sysbench-1.0.3/src/sb_counter.h:68: undefined reference to `ck_pr_store_64'
db_driver.o: In function `sb_counter_val':
/builddir/build/BUILD/sysbench-1.0.3/src/sb_counter.h:62: undefined reference to `ck_pr_load_64'
db_driver.o: In function `sb_counter_inc':
/builddir/build/BUILD/sysbench-1.0.3/src/sb_counter.h:68: undefined reference to `ck_pr_store_64'
sb_histogram.o: In function `merge_intermediate_into_cumulative':
/builddir/build/BUILD/sysbench-1.0.3/src/sb_histogram.c:211: undefined reference to `ck_pr_fas_64'
/builddir/build/BUILD/sysbench-1.0.3/src/sb_histogram.c:211: undefined reference to `ck_pr_fas_64'
/builddir/build/BUILD/sysbench-1.0.3/src/sb_histogram.c:211: undefined reference to `ck_pr_fas_64'
/builddir/build/BUILD/sysbench-1.0.3/src/sb_histogram.c:211: undefined reference to `ck_pr_fas_64'
/builddir/build/BUILD/sysbench-1.0.3/src/sb_histogram.c:211: undefined reference to `ck_pr_fas_64'
sb_histogram.o:/builddir/build/BUILD/sysbench-1.0.3/src/sb_histogram.c:211: more undefined references to `ck_pr_fas_64' follow
sb_histogram.o: In function `sb_histogram_update':
/builddir/build/BUILD/sysbench-1.0.3/src/sb_histogram.c:115: undefined reference to `ck_pr_inc_64'
sb_histogram.o: In function `sb_histogram_get_pct_intermediate':
/builddir/build/BUILD/sysbench-1.0.3/src/sb_histogram.c:145: undefined reference to `ck_pr_fas_64'
/builddir/build/BUILD/sysbench-1.0.3/src/sb_histogram.c:145: undefined reference to `ck_pr_fas_64'
/builddir/build/BUILD/sysbench-1.0.3/src/sb_histogram.c:145: undefined reference to `ck_pr_fas_64'
/builddir/build/BUILD/sysbench-1.0.3/src/sb_histogram.c:145: undefined reference to `ck_pr_fas_64'
/builddir/build/BUILD/sysbench-1.0.3/src/sb_histogram.c:145: undefined reference to `ck_pr_fas_64'
sb_histogram.o:/builddir/build/BUILD/sysbench-1.0.3/src/sb_histogram.c:145: more undefined references to `ck_pr_fas_64' follow
sb_counter.o: In function `sb_counter_val':
/builddir/build/BUILD/sysbench-1.0.3/src/sb_counter.h:62: undefined reference to `ck_pr_load_64'

Full i686 build log : https://kojipkgs.fedoraproject.org//work/tasks/6499/18256499/build.log

Regards,
Xavier

**Comment by akopytov** [comment]
Thanks for reporting. The underlying issue is that i686 does not provide 64-bit atomics, and applications are supposed to test ConcurrencyKit feature macros and adjust data types accordingly at the compilation stage.

I'm not sure why it works with bundled CK on i686 -- it really shouldn't, and that probably indicates some issues with either sysbench build system or build environment on your end.

Anyway, fixing this will require somewhat invasive changes to sysbench, I'm going to look into it shortly.

**Comment by akopytov** [comment]
This should be fixed in release 1.0.4.

**Comment by xavierba** [comment]
Builds fine on i686 now, thanks for the fix.

**Comment by kgizdov** [comment]
I'm having the exact same error when trying to build on armv6h (RaspberryPi 2).

**Comment by akopytov** [comment]
@kgizdov I don't have an armv6h machine around. Can you please check if the following patch fixes the build for you?

``` diff
diff --git a/src/sb_ck_pr.h b/src/sb_ck_pr.h
index b782cd9..e3ef976 100644
--- a/src/sb_ck_pr.h
+++ b/src/sb_ck_pr.h
@@ -63,7 +63,7 @@
 
 #include "ck_pr.h"
 
-#if defined(__i386__) && !defined(CK_F_PR_LOAD_64)
+#if !defined(CK_F_PR_LOAD_64)
 
 #ifndef __GNUC__
 #  error Unsupported platform

```

**Comment by kgizdov** [comment]
yes, that seems to have done the trick

**Comment by akopytov** [comment]
Pushed the fix to both 1.0 and master branches. Thanks!
